prow/config/config.go

```diff
@@ -46,12 +47,12 @@ type Config struct {
 	// Periodics are not associated with any repo.
 	Periodics []Periodic `json:"periodics,omitempty"`
 
-	Tide             Tide             `json:"tide,omitempty"`
-	Plank            Plank            `json:"plank,omitempty"`
-	Sinker           Sinker           `json:"sinker,omitempty"`
-	Deck             Deck             `json:"deck,omitempty"`
-	BranchProtection BranchProtection `json:"branch-protection,omitempty"`
```

Comment:
We avoid embedding structs throughout this file and some other parts of Prow where it doesn't make sense for the parent struct to take on the functions of the child struct (if any functions exist).
If we add functions to the BranchProtection or PushGateway structs they shouldn't be exposed from the Config struct in order to maintain encapsulation and avoid conflicting function names.

Reply:
Done